common/src/main/java/io/netty/util/internal/PlatformDependent.java

```diff
@@ -668,14 +684,110 @@ private static boolean isAndroid0() {
         return android;
     }
 
+    private static boolean isOs(String os) {
+        return SystemPropertyUtil.get("os.name", "").toLowerCase(Locale.US).contains(os);
+    }
+
     private static boolean isWindows0() {
-        boolean windows = SystemPropertyUtil.get("os.name", "").toLowerCase(Locale.US).contains("win");
+        boolean windows = isOs("win");
         if (windows) {
             logger.debug("Platform: Windows");
         }
         return windows;
     }
 
+    private static boolean isLinux0() {
+        boolean linux = isOs("linux");
+        if (linux) {
+            logger.debug("Platform: Linux");
+        }
+        return linux;
+    }
+
+    private static boolean isOsx0() {
+        boolean osx = isOs("os x") || isOs("osx");
+        if (osx) {
+            logger.debug("Platform: OSX");
+        }
+        return osx;
+    }
+
+    private static int cacheLineSize0() {
+        return AccessController.doPrivileged(new PrivilegedAction<Integer>() {
+            @Override
+            public Integer run() {
+                if (IS_LINUX) {
+                    BufferedReader in = null;
+                    try {
+                        in = new BufferedReader(new InputStreamReader(new FileInputStream("/proc/cpuinfo")));
+                        String line;
+                        while ((line = in.readLine()) != null) {
+                            if (line.startsWith("cache_alignment")) {
+                                String[] parts = StringUtil.split(line, ':');
+                                return Integer.parseInt(parts[1].trim());
+                            }
+                        }
+                    } catch (Throwable cause) {
+                        logger.debug("Failed to retrieve cache line size", cause);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "Failed to retrieve cache line size" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1192)